optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/ScoreDirectorFactoryFactory.java

```diff
@@ -180,7 +188,17 @@ private void checkMultipleScoreDirectorFactoryTypes(EasyScoreDirectorFactory eas
         }
     }
 
-    protected ConstraintStreamScoreDirectorFactory<Solution_, Score_> buildConstraintStreamScoreDirectorFactory(
+    private void enforceNoAlphaNetworkCompiler() {
```

Comment:
Fail-fast.

Reply:
We typically call such methods "validate"

```suggestion
    private void validateNoAlphaNetworkCompiler() {
```

Reply:
Looks like this validation doesn't trigger if BAVET is used and it's true

Reply:
Maybe, you can just create a validateDroolsAlphaNetworkCompilation method that is always called, not just when it needs to be disabled?